pkg/operator/starter.go

```diff
@@ -38,19 +39,30 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 
 	// Create Clients
 	c := clients.NewBuilder(opConfig.UserAgent, string(opConfig.CSIDriverName), controllerConfig, resync).
-		WithHyperShiftGuest(guestKubeConfigString).
+		WithHyperShiftGuest(guestKubeConfigString, opConfig.CloudConfigNamespace).
 		BuildOrDie(ctx)
 
+	klog.Infof("Building clients is done")
+
+	// Build ControllerConfig
+	csiOperatorControllerConfig, err := opConfig.OperatorControllerConfigBuilder(ctx, flavour, c)
+	if err != nil {
+		klog.Errorf("error building operator config: %v", err)
+		return err
+	}
+
 	// Load generated assets.
 	assetDir := filepath.Join(opConfig.AssetDir, string(flavour))
 	a, err := generated_assets.NewFromAssets(assets.ReadFile, assetDir)
 	if err != nil {
 		return err
 	}
-	a.SetReplacements(operator.DefaultReplacements(controlPlaneNamespace))
+	defaultReplacements := operator.DefaultReplacements(controlPlaneNamespace)
+	if csiOperatorControllerConfig.ExtraReplacementsFunc != nil {
+		defaultReplacements = append(defaultReplacements, csiOperatorControllerConfig.ExtraReplacementsFunc()...)
+	}
```

Comment:
Similar idea here, could we require each platform to define a replacement func? It could just return the value of DefaultReplacements() if there is nothing more to add.